File: esports-backend/src/main/java/com/esports/controller/MarketplaceController.java
package com.esports.controller;

import com.esports.dto.MarketplaceServiceDTO;
import com.esports.service.MarketplaceService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api")
@CrossOrigin(origins = "http://localhost:5173")
public class MarketplaceController {

    @Autowired
    private MarketplaceService marketplaceService;

    @GetMapping("/services")
    public ResponseEntity<?> getAllServices() {
        try {
            List<MarketplaceServiceDTO> services = marketplaceService.getAllActiveServices();
            return ResponseEntity.ok(services);
        } catch (Exception e) {
            return ResponseEntity.status(500).body(Map.of("error", "Failed to get services"));
        }
    }
}

File: Upkart1/Controllers/MarketplaceController.cs
using Microsoft.AspNetCore.Mvc;
using Upkart1.DTO;
using Upkart1.Services;

namespace Upkart1.Controllers
{
    [ApiController]
    [Route("api")]
    public class MarketplaceController : ControllerBase
    {
        private readonly IMarketplaceService _marketplaceService;

        public MarketplaceController(IMarketplaceService marketplaceService)
        {
            _marketplaceService = marketplaceService;
        }

        [HttpGet("services")]
        public IActionResult GetAllServices()
        {
            try
            {
                var services = _marketplaceService.GetAllActiveServices();
                return Ok(services);
            }
            catch (Exception)
            {
                return StatusCode(500, new { error = "Failed to get services" });
            }
        }
    }
}
